### DISCUSSION TITLE
React components

### DISCUSSION THREAD

**Discussion by Phoenixrr2113** [LiveDuo/destack#72] [Q&A]
Are you goals eventually to be able to drag and drop react components in the editor? Also can I see a list of your goals? I want to help make this thing big. you probably cant tell but I'm super excited about your project and I have lots of ideas features to build.

**Comment by LiveDuo** [comment]
Hey @Phoenixrr2113,

Glad you find Destack interesting.

There isn't a roadmap right now. The reason is that most of the previous tasks are close to completion. These tasks were migrating the editor to Craft.js, better typescript support and e2e tests, multi page support, multiple themes and react support.

There are many ideas suggested from Destack users (and some from me). 

Here are some:
- Support React components
- Support custom HTML
- Support dynamic content (such as blog posts)
- Improve primary color support
- Use grapesjs plugins (for the grapes version)
- Admin UI support (in production)
- Even more themes/components
- Custom Tailwind css config

There could be more that I can't remember right now.

I'm not sure what the next step is exactly but it's probably work on a few rough edges and stabilise the current version (as it comes from a big rewrite of the core logic).

What comes next is probably dynamic content. It has quite a few nuances that I need to figure out before starting working on it.

**Comment by thanhdevapp** [comment]
The future it will be like ? https://landing.ant.design/ Plans like integration with a backend like strapi.

**Reply by LiveDuo** [reply]
@thanhdevapp 

Possibly Strapi integration along with any other backend will come with "Dynamic content support".

The plan is to allow data to be passed on the components through Next.js `getInitialProps`, `getServerSideProps` or even Next.js API Routes so for a bit of code one would be able to integrate their favourite backend solution.
